tests/Resources/Page/ODH/JupyterHub/HighAvailability.robot

```diff
@@ -0,0 +1,60 @@
+*** Settings ***
+Library  Collections
+
+*** Keywords ***
+Verify Deployment
+    [Arguments]  ${component}  ${nPods}  ${nContainers}  ${containerNames}
+    #No. of replicas
+    Length Should Be  ${component}  ${nPods}
+
+    FOR  ${index}  IN RANGE  0  ${nPods}
+        &{pod} =  Set Variable  ${component}[${index}]
+        #No. of containers
+        Length Should Be  ${pod.status.containerStatuses}  ${nContainers}
+        @{names} =  Create List
+        FOR  ${j}  IN RANGE  0  ${nContainers}
+            Append To List  ${names}  ${pod.status.containerStatuses[${j}].name}
+            Should Be Equal As Strings  ${pod.status.phase}  Running
+            ${state} =  Get Dictionary Keys  ${pod.status.containerStatuses[${j}].state}
+            Should Be Equal As Strings  ${state}[0]  running
+        END
+        Sort List  ${names}
+        Sort List  ${containerNames}
+        Lists Should Be Equal  ${names}  ${containerNames}
+    END
+
+Verify JupyterHub Deployment
+    [Arguments]  ${component}  ${nPods}  ${nContainers}  ${containerNames}
+    #Standard deployment check
+    Verify Deployment  ${component}  ${nPods}  ${nContainers}  ${containerNames}
+
+    ${leader} =  Set Variable  None
+    ${leader-found} =  Set Variable  False
+    #Force to integer
+    ${standby} =  Set Variable  ${0}
+
+    FOR  ${index}  IN RANGE  0  ${nPods}
+        &{pod} =  Set Variable  ${component}[${index}]
+        FOR  ${j}  IN RANGE  0  ${nContainers}
+            IF  '${pod.status.containerStatuses[${j}].name}' == 'jupyterhub'
+                #leader's pod is recognized by jupyterhub container in ready status
+                IF  '${pod.status.containerStatuses[${j}].ready}' == 'True'
+                    IF  ${leader-found}
```

Comment:
I think the code could be simplified using the same logic as standby `${leader} =  Set Variable  ${leader+1}` and assert if not equals to 1?

Reply:
I don't think it would simplify things by a lot. The original idea I had was to use the `${leader}` variable to capture the actual pod name, thus being able to reuse this keyword down the line when implementing a TC for HA functionality (i.e.
```
kill ${leader}
Do HA test stuff
Verify Jupyterhub Deployment  #Check a new leader has been elected
```
etc.